python/run_tests.py

```diff
@@ -0,0 +1,538 @@
+                            assert " = " in line, "The string ' = ' is present " +\
+                                                  "in the string: " + line
+                            line = line[:line.index(" = ") + 3]
+                            line += "{0}".format(value)
+                            break
+                modified_lines.append(line)
+
+        with open(modified_fname, "w") as modified_file:
+            modified_file.write("\n".join(modified_lines))
+            modified_file.write("\n")
+
+    def _setup_parameter_mapping(self):
+        raise NotImplementedError("You are calling a purely virtual method.")
+
+    def _process_output(self, output):
+        raise NotImplementedError("You are calling a purely virtual method.")
+
+    @property
+    def name(self):
+        return self._application_name
+
+    def run(self, n_processes=1, reference_result=False):
+        # current directory
+        cwd = os.getcwd()
+        # directory checks
+        application_dir = path.join(cwd, "applications")
+        prm_fname = path.join(application_dir,
+                              self._application_name + self._prm_suffix)
+        exec_fname = path.join(application_dir, self._application_name)
+        if not path.exists(application_dir) or not path.exists(exec_fname)\
+            or not path.exists(prm_fname):
+            raise ApplicationNotExistingError()
+        # create dump directory
+        dump_dir = path.join(cwd, "dump")
+        if not path.exists(dump_dir):
+            os.mkdir(dump_dir)
+        # setup problem specific parameters
+        self._setup_parameter_mapping()
+        # setup parameter
+        modified_prm_fname = path.join(dump_dir,
+                                       self._application_name + self._prm_suffix)
+        self._setup_parameter_file(prm_fname, modified_prm_fname)
+        assert path.exists(modified_prm_fname)
+        # create symbolic link
+        os.symlink(exec_fname, path.join(dump_dir, self._application_name))
+        # perform convergence test
+        os.chdir(dump_dir)
+        print("Running convergence test " + self._application_name + "...",
+              end="", flush=True)
+        output = subprocess.check_output(["mpirun", "-np", str(n_processes),
+                                          "./" + self._application_name], text=True)
+        print("   done!")
+        # finalize
+        os.chdir(cwd)
+        subprocess.run(["rm", "-rf", dump_dir], check=True)
+        # compare convergence table
+        success = self._process_output(output, reference_result)
+        assert isinstance(success, bool)
+        return success
+
+
+class ThermalTGVConvergenceTest(ConvergenceTest):
+    def __init__(self, peclet_number, test_type=TestType.temporal,
+                 n_spatial_cycles=2, n_temporal_cycles=2, n_initial_global_refinements=3,
+                 n_global_refinements=3, initial_time_step=1e-1, final_time_step=1e-3):
```

Comment:
The amount of arguments could be reduced to 5, e.g. 
```
def __init__(self, number, test_type=TestType.temporal, n_cycles, n_global_refinements, time_step)
```
What is to be done with the last two can be determined with `test_type`. Just an idea.